src/main/java/pl/onsight/wypozyczalnia/controller/AdminController.java

```diff
@@ -32,29 +35,64 @@ public ModelAndView adminPage(ModelAndView modelAndView) {
         return modelAndView;
     }
 
+    @DeleteMapping(path = "/product-order/{id}")
+    @ResponseBody
+    public void removeProductOrder(@PathVariable Long id) {
+        productOrderService.removeProductOrder(id);
+    }
+
+    @DeleteMapping(path = "/product/{id}")
+    @ResponseBody
+    public void removeProduct(@PathVariable Long id) {
+        productService.removeProduct(id);
+    }
+
+
     @PostMapping("/admin-page")
     public ModelAndView addProduct(@ModelAttribute @Valid ProductEntity product,
                                    BindingResult bindingResult, ModelAndView modelAndView) {
+        List<ProductEntity> allProducts = productService.findAllProducts();
+        List<String> allNamesOfProducts = allProducts.stream().map(name -> name.getProductName()).collect(Collectors.toList());
         if (bindingResult.hasErrors()) {
             modelAndView.setViewName("admin");
+        } else if (allNamesOfProducts.contains(product.getProductName())) {
+            modelAndView.addObject("info", new Info("Produkt o nazwie " + product.getProductName() + " już istnieje!", false));
+            return adminPage(modelAndView);
```

Comment:
mozna od rayzu druga liste zrobic, pierwsza jet niepotrzebna i a lista, i warunek do walidatora i cacy, i po dodaniu produktu tez jakies info powinno byc ze spoko sie doalo, ale to peirdola i to mozna pozniej, w ogoel byloby sopko gdyby przy probie zapisu z jakas zajeta nazwa sie nie usuwaly wypelnione pola, ale to taka luzna mysl na kiedys.

Reply:
a i doszlo nowe pole w produkcie, wiec wypada je dodac w formularzu

Reply:
powyszy koment w mocy pozostal :)